geonode/qgis_server/tests/test_views.py

```diff
@@ -133,6 +133,29 @@ def test_ogc_specific_layer(self):
         self.assertEqual(response.get('Content-Type'), 'image/tiff')
         self.assertEqual(what('', h=response.content), 'tiff')
 
+        # Layer is already on the database
+        # checking the Link
+        links = uploaded.link_set.download().filter(
+            name__in=settings.DOWNLOAD_FORMATS_RASTER)
+
+        for link in links:
+            if "download-qlr" in link.url:
+                self.assertTrue("download-qlr" in link.url)
+            if "download-qgs" in link.url:
+                self.assertTrue("download-qgs" in link.url)
+
+        # QLR
+        response = self.client.get(
+            reverse('qgis_server:download-qlr', kwargs=params))
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.get('Content-Type'), 'application/xml')
```

Comment:
Also test filename extension if possible.

Can you also change the content-type to follow QGIS MIME:
https://github.com/qgis/QGIS/blob/master/rpm/sources/qgis-mime.xml

With this, the corresponding MIME type will be:

QLR: application/x-qgis-layer-definition
QGS: application/x-qgis-project

Hopefully, this will fix the problem in Safari.

CC @timlinux @Gustry , I hope this point to the correct info from QGIS.

Reply:
Thanks ! I wasn't aware.
I see it's a subclass of `application/xml`. Let's try.

Reply:
the trick worked @lucernae 
thanks !